onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlSTFT.h

```diff
@@ -0,0 +1,610 @@
+                m_framingOperator.op.Get(),
+                m_framingOperator.persistentResourceBinding ? &*m_framingOperator.persistentResourceBinding : nullptr,
+                gsl::make_span(initializationInputBindings)
+            ));
+        }
+
+        auto execBindingProps = m_framingOperator.op->GetBindingProperties();
+
+        D3D12_DESCRIPTOR_HEAP_DESC descriptorHeapDesc = {};
+        descriptorHeapDesc.Flags = D3D12_DESCRIPTOR_HEAP_FLAG_SHADER_VISIBLE;
+        descriptorHeapDesc.NumDescriptors = execBindingProps.RequiredDescriptorCount;
+        descriptorHeapDesc.Type = D3D12_DESCRIPTOR_HEAP_TYPE_CBV_SRV_UAV;
+
+        ORT_THROW_IF_FAILED(m_d3dDevice->CreateDescriptorHeap(&descriptorHeapDesc, IID_GRAPHICS_PPV_ARGS(&m_framingOperator.descriptorHeap)));
+
+        DML_BINDING_TABLE_DESC bindingTableDesc = {};
+        bindingTableDesc.Dispatchable = m_framingOperator.op.Get();
+        bindingTableDesc.CPUDescriptorHandle = m_framingOperator.descriptorHeap->GetCPUDescriptorHandleForHeapStart();
+        bindingTableDesc.GPUDescriptorHandle = m_framingOperator.descriptorHeap->GetGPUDescriptorHandleForHeapStart();
+        bindingTableDesc.SizeInDescriptors = execBindingProps.RequiredDescriptorCount;
+
+        ORT_THROW_IF_FAILED(m_dmlDevice->CreateBindingTable(&bindingTableDesc, IID_PPV_ARGS(&m_framingOperator.bindingTable)));
+
+        ORT_THROW_IF_FAILED(m_dmlDevice->CreateCommandRecorder(IID_PPV_ARGS(&m_framingOperator.commandRecorder)));
+    }
+
+    STDMETHOD(Compute)(IMLOperatorKernelContext* context)
+    {
+        try
+        {
+            ComPtr<IUnknown> executionObject;
+            context->GetExecutionInterface(executionObject.GetAddressOf());
+
+            ComPtr<ID3D12GraphicsCommandList> commandList;
+            ORT_THROW_IF_FAILED(executionObject.As(&commandList));
+
+            ComPtr<ID3D12Resource> framingOutputResource;
+            ORT_THROW_IF_FAILED(context->AllocateTemporaryData(m_framingOperator.outputBufferSizeInBytes, &framingOutputResource));
+
+            DispatchFramingOperator(commandList.Get(), context, framingOutputResource.Get());
+
+            ComPtr<ID3D12Resource> outputResource = DmlSTFTHelpers::GetOutputResourceFromKernelContext(context, 0);
+
+            D3D12_RESOURCE_BARRIER uavBarrier = { CD3DX12_RESOURCE_BARRIER::UAV(nullptr) };
+            commandList->ResourceBarrier(1, &uavBarrier);
+
+            return m_dftOperator.op->Compute(
+                commandList.Get(),
+                context,
+                framingOutputResource.Get(),
+                m_dftOperator.inputDims,
+                outputResource.Get(),
+                m_dftOperator.outputDims,
+                m_dftOperator.dftLength
+            );
+        }
+        catch (...)
+        {
+            return E_FAIL;
+        }
+
+        return S_OK;
+    }
+
+    void DispatchFramingOperator(ID3D12GraphicsCommandList* commandList, IMLOperatorKernelContext* context, ID3D12Resource* outputResource)
```

Comment:
>DispatchFramingOperator [](http://example.com/codeflow?start=9&length=23)

I'm glad someone on our team knows this part with all the binding steps, because I wouldn't be able to write this without heavily looking at existing code, and then still probably getting it wrong 😅.